tests/h/services/group_test.py

```diff
@@ -62,6 +64,59 @@ def test_it_returns_None_if_no_matching_group(self, svc):
         assert svc.fetch_by_groupid("group:rando@dando.com") is None
 
 
+class TestFilter(object):
+    def test_it_filters_by_name(self, svc, groups):
+        filters = {"name": "Hello"}
+
+        filtered_groups = svc.filter(**filters)
+
+        assert len(filtered_groups.all()) == 1
+        assert filtered_groups[0].name == "Hello"
+
+    def test_it_returns_all_groups_if_no_filters(self, svc, groups):
+        filtered_groups = svc.filter()
+
+        # results include public group in addition to ``groups``
+        assert len(filtered_groups.all()) == len(groups) + 1
+
+    def test_it_ignores_unknown_filters(self, svc, groups):
```

Comment:
The `groups` argument on each of these test methods is triggering pylint's "unused argument" warning. If a test body isn't going to use a fixture, but you just want that fixture to be called before the test is run, you should do `@pytest.mark.usefixtures("groups")`. And given that every test method in this class uses the `groups` fixture you could just put the `usefixtures` decorator once on the class instead of on each method.

Reply:
Yeah, totally remedial. Will fix.

Reply:
@seanh Oddly, I cannot get this to work! The tests all fail if I use `@pytest.mark.use_fixtures` on the whole class. I tried moving the fixture outside of the class itself and a couple of other things. I don't know if it has something to do with persistence or...some sort of test config, but I'm unable to get these tests to pass without passing the `groups` fixture directly to each test. Any ideas?

Reply:
Try this:

```diff
diff --git a/tests/h/services/group_test.py b/tests/h/services/group_test.py
index 84582e1b5..b3d2abbae 100644
--- a/tests/h/services/group_test.py
+++ b/tests/h/services/group_test.py
@@ -64,8 +64,9 @@ class TestGroupServiceFetchByGroupid(object):
         assert svc.fetch_by_groupid("group:rando@dando.com") is None
 
 
+@pytest.mark.usefixtures("groups")
 class TestFilter(object):
-    def test_it_filters_by_name(self, svc, groups):
+    def test_it_filters_by_name(self, svc):
         filters = {"name": "Hello"}
 
         filtered_groups = svc.filter(**filters)
@@ -79,7 +80,7 @@ class TestFilter(object):
         # results include public group in addition to ``groups``
         assert len(filtered_groups.all()) == len(groups) + 1
 
-    def test_it_ignores_unknown_filters(self, svc, groups):
+    def test_it_ignores_unknown_filters(self, svc):
         filters = {"name": "Hello", "foo": "bar"}
 
         filtered_groups = svc.filter(**filters)
@@ -87,19 +88,19 @@ class TestFilter(object):
         assert len(filtered_groups.all()) == 1
         assert filtered_groups[0].name == "Hello"
 
-    def test_name_filter_is_case_insensitive(self, svc, groups):
+    def test_name_filter_is_case_insensitive(self, svc):
         filters = {"name": "Amber"}
         filtered_groups = svc.filter(**filters)
 
         assert len(filtered_groups.all()) == 2
 
-    def test_name_filter_is_wildcarded(self, svc, groups):
+    def test_name_filter_is_wildcarded(self, svc):
         filters = {"name": "Finger"}
         filtered_groups = svc.filter(**filters)
 
         assert len(filtered_groups.all()) == 2
 
-    def test_results_sorted_by_created_desc(self, svc, groups):
+    def test_results_sorted_by_created_desc(self, svc):
         filters = {"name": "Finger"}
         filtered_groups = svc.filter(**filters)
 
```

Reply:
I tried exactly that and all the tests fail.

Reply:
Oh, crud! I see my damage. Typo in `usefixtures`; I was trying `use_fixtures`. Unrecognized `mark` things just get ignored, so...